lxd/storage/backend_lxd.go

```diff
@@ -2547,8 +2548,8 @@ func (b *lxdBackend) UpdateCustomVolume(projectName string, volName string, newD
 
 // UpdateCustomVolumeSnapshot updates the description of a custom volume snapshot.
 // Volume config is not allowd to be updated and will return an error.
-func (b *lxdBackend) UpdateCustomVolumeSnapshot(projectName string, volName string, newDesc string, newConfig map[string]string, op *operations.Operation) error {
-	logger := logging.AddContext(b.logger, log.Ctx{"project": projectName, "volName": volName, "newDesc": newDesc, "newConfig": newConfig})
+func (b *lxdBackend) UpdateCustomVolumeSnapshot(projectName string, volName string, newDesc string, newConfig map[string]string, newExpiryDate time.Time, op *operations.Operation) error {
```

Comment:
@monstermunchkin should the expiry time come via the `newConfig` map rather than a separate argument? The conversion from string to time.Time can be done inside the function.

Reply:
I guess it would be possible. With instances however, it's not a config key either. That's why I chose to do it this way.

Reply:
@monstermunchkin ah, no, I see now, the config property `snapshots.expiry` is *only* a config property of the parent volume, but that sets the ExpiresAt property of each snapshot (so the snapshots themselves don't have the `snapshots.expiry` property (well they do, because the config is snapshotted as well from the parent volume, but it is not used).

Instead, at snapshot create time, the parent volumes `snapshots.expiry` setting is used to calculate an ExpiresAt absolute time in the future, which the SQL queries use to select snapshot volumes to be removed.